AgentDiscoveries-Frontend/app/src/components/login.js

```diff
@@ -0,0 +1,123 @@
+import * as React from "react";
+import {
+    Form,
+    FormGroup,
+    FormControl,
+    Button,
+    ControlLabel
+} from "react-bootstrap";
+import * as CRUD from "./crud"
+
+export default class Login extends React.Component {
+
+    constructor() {
+        super();
+        this.state = {
+            authenticationMessage: "",
+            isLoggedInMessage: "",
+        }
+    }
+
+    componentDidMount() {
+        this.updateIsLoggedIn();
+    }
+
+    render() {
+        return (
+            <div>
+                <div>{this.state.isLoggedInMessage}</div>
+                <Button onClick={this.logOut.bind(this)}>Log out</Button>

```

Comment:
The binding issue is an annoyance that often comes up in react. There are a couple of other ways of writing this, if you like either of them better then feel free to change, but I think it's fine as is:
# Bind them all in the constructor:
`this.logOut = this.logOut.bind(this);`
And then when you get to the rendering you can just use this.logOut

# Arrow functions to wrap them
`() => this.logOut()`